### PR TITLE
add lycon

### LINKED ISSUE #12555: Package lycon
https://github.com/ethereon/lycon

### PR DESCRIPTION
**Pull request by carlodri**
https://github.com/ethereon/lycon

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Closes #12555

### CODE REVIEW

**Comment by carlodri** [comment]
@conda-forge/staged-recipes how should I deal with the fact that the package has split licenses (https://github.com/ethereon/lycon/blob/master/LICENSE)?

**Comment by carlodri** [comment]
@conda-forge/help-python I'm lost here as usual with these conda package conflict descriptors that are totally obscure to me.... can you give me any help please?

File: recipes/lycon/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake >=2.8
+  host:
+    - python
+    - pip
+    - numpy
```
Comment (chrisburr) [inline recipes/lycon/meta.yaml:23]: Try adding this temporarily:
```suggestion
    # TODO: Remove the version constraint and re-render once this is merged into a feedstock
    - numpy 1.16
```

Comment (carlodri) [reply recipes/lycon/meta.yaml]: Thanks for stepping in @chrisburr, sadly it seems like "downgrading" `numpy` did not help... Any other idea?

Comment (chrisburr) [reply recipes/lycon/meta.yaml]: Linux is now working isn't it?

Comment (carlodri) [reply recipes/lycon/meta.yaml]: oh yes, sorry, you're right, I rushed my conclusion.... let me check the CI

Comment (carlodri) [reply recipes/lycon/meta.yaml]: so, OSX seems to fail the import test silently, whereas on windows cmake complains that the PYTHON_LIB_PATH variable is not found/set

**Comment by carlodri** [comment]
ok, so failure on windows is due to this cmake thing;

https://github.com/ethereon/lycon/blob/93076d39ca8e42a509af471d620daa859c6c2a3f/cmake/LocatePythonLibs.cmake#L19-L20

which does not work on windows.

**Pull request closed without merging**